src/Microsoft.OData.Core/Uri/NodeToStringBuilder.cs

```diff
@@ -470,7 +470,88 @@ internal String TranslateNode(QueryNode node)
         internal String TranslateFilterClause(FilterClause filterClause)
         {
             Debug.Assert(filterClause != null, "filterClause != null");
-            return this.TranslateNode(filterClause.Expression);
+            string translatedNode = this.TranslateNode(filterClause.Expression);
+
+            switch (filterClause.Expression.Kind)
+            {
+                // $expand=Items($filter=$it/Product eq 'Some Product')
+                case QueryNodeKind.BinaryOperator:
+                    BinaryOperatorKind binaryNodeKind = (filterClause.Expression as BinaryOperatorNode).OperatorKind;
+
+                    string binaryOperator = this.BinaryOperatorNodeToString(binaryNodeKind);
+                    string[] binarySeparator = { binaryOperator };
+                    string[] binarySubstrings = translatedNode.Trim().Split(binarySeparator, StringSplitOptions.RemoveEmptyEntries);
+                    string leftBinary = binarySubstrings[0];
+                    string rightBinary = binarySubstrings[1];
+
+                    SingleValuePropertyAccessNode leftBinaryNode = ((filterClause.Expression as BinaryOperatorNode).Left as ConvertNode)?.Source as SingleValuePropertyAccessNode;
+                    SingleValuePropertyAccessNode rightBinaryNode = ((filterClause.Expression as BinaryOperatorNode).Right as ConvertNode)?.Source as SingleValuePropertyAccessNode;
+
+                    if (leftBinaryNode != null && IsDifferentSource(filterClause, leftBinaryNode))
+                    {
+                        leftBinary = "$it/" + leftBinary;
+                    }
+
+                    if (rightBinaryNode != null && IsDifferentSource(filterClause, rightBinaryNode))
+                    {
+                        rightBinary = "$it/" + rightBinary;

```

Comment:
I am worry about the implementation here, does it only work for one level? I mean $it/Product?  Do we work:  $it/Address/City

Reply:
It should support `$it/Address/City`
I will add tests to validate that.